Snakefile

```diff
@@ -213,12 +293,14 @@ rule add_h5_clade:
     message: "Adding in a column for h5 clade numbering"
     input:
         metadata = "results/{subtype}/metadata.tsv",
-        clades_file = files.clades_file
+        clades_file = resolve_config_path(files.clades_file)
     output:
         metadata= "results/{subtype}/metadata-with-clade.tsv"
+    params:
+        script = os.path.join(workflow.current_basedir, "clade-labeling/add-clades.py")
```

Comment:
I don't think this makes sense. It's not a config path, and I don't think it should be implied to be one or go thru that resolution process (esp. if forced). If we want to share checks and error handling, we can refactor those out into a separate function.

Reply:
> I don't think this makes sense. 

👍 

> It's not a config path, and I don't think it should be implied to be one or go thru that resolution process (esp. if forced). 

I almost suggested renaming the method to `resolve_path()` in anticipation of this objection. 

> If we want to share checks and error handling, we can refactor those out into a separate function.

My broader point here was this: it's path munging, we know that paths are already a big source of confusion, and munging them is going to amplify that. Good error messages can help.

Reply:
Actually, thinking about this a bit more, I think this particular case would work just fine with the stock `resolve_config_path()` — that would allow somebody to supply an analysis-specific version of this script without requiring the writing of a whole new custom Snakemake rule to replace this one with; the default path resolution order would fall back to the current_basedir anyway if a custom script wasn't present. 

(I do expect somebody is going to show up with a plausible reason for wanting to change the path resolution ordering in some cases but that can stay in YAGNI land until it actually happens…)

Reply:
I still disagree. Anything we expose for overriding in an working analysis directory is part of the workflow's _public interface_. I don't think we want to place internal programs into that interface.

Reply:
I think the whole thing is already effectively the public interface if we're trying to enable other folks to run it. Additionally, I think once we get used to the affordance of "run a workflow multiple times into distinct analysis directories and tweak specific bits", we will run into cases where the variable being tweaked is the behavior of one of these scripts.

Agree to disagree, I guess.